plugin/Field/SearchcriteriaFields.php

```diff
@@ -108,8 +108,11 @@ public function getFieldLabelsOfInputs(array $inputFormFields): array
 		$output = [];
 		$pGeoPosition = new GeoPosition;
 		$geoRangeFields = array_flip($pGeoPosition->getSearchCriteriaFields());
-
 		foreach ($inputFormFields as $name => $value) {
+			if (in_array($name, ['message'])) {
```

Comment:
I think, instead of adding new ifs and bool parameters, it should be found out why `$pFieldsCollection->getFieldByModuleAndName()` doesn't know the `message` field anymore

Reply:
Hi @jayay. I do not understand. Because the `message` field is not returned by the API. Which module do I need to add the `message` field to?

Reply:
Those were usually added to the `''` (empty string) module.

Reply:
I add a message field to the search criteria module. But when submitting the form, there will be an error. Because it will call CreateSearchCriteria API and will give error no message

Reply:
This change worked for me:

```diff
diff --git a/plugin/Field/SearchcriteriaFields.php b/plugin/Field/SearchcriteriaFields.php
index 57754ef..87c362b 100644
--- a/plugin/Field/SearchcriteriaFields.php
+++ b/plugin/Field/SearchcriteriaFields.php
@@ -102,29 +102,24 @@ class SearchcriteriaFields
                        ->addFieldsSearchCriteria($pFieldsCollection)
                        ->addFieldsAddressEstate($pFieldsCollection);

-               $pGeoFieldsCollection = new FieldModuleCollectionDecoratorGeoPositionFrontend(new FieldsCollection);
-               $pFieldsCollection->merge($pGeoFieldsCollection);
+               $pFieldsCollection->merge(new FieldModuleCollectionDecoratorGeoPositionFrontend(new FieldsCollection));
+               $pFieldsCollection->merge(new FieldModuleCollectionDecoratorFormContact(new FieldsCollection));

                $output = [];
                $pGeoPosition = new GeoPosition;
                $geoRangeFields = array_flip($pGeoPosition->getSearchCriteriaFields());
                foreach ($inputFormFields as $name => $value) {
-                       if (in_array($name, ['message'])) {
-                               $output[$name] = $value;
-                               continue;
-                       }
                        $aliasedFieldName = $this->getFieldNameOfInput($name);
                        if (in_array($name, $pGeoPosition->getSearchCriteriaFields())) {
                                $aliasedFieldName = $geoRangeFields[$name];
                        }
-                       $module = onOfficeSDK::MODULE_ESTATE;
-                       $pField = $pFieldsCollection->getFieldByModuleAndName($module, $aliasedFieldName);
+                       $pField = $pFieldsCollection->getFieldByKeyUnsafe($aliasedFieldName);

                        if (FieldTypes::isRangeType($pField->getType()))
                        {
-                               if (stristr($name, self::RANGE_FROM)) {
+                               if (__String::getNew($name)->endsWith(self::RANGE_FROM)) {
                                        $output[$pField->getLabel().' (min)'] = $value;
-                               } elseif (stristr($name, self::RANGE_UPTO)) {
+                               } elseif (__String::getNew($name)->endsWith(self::RANGE_UPTO)) {
                                        $output[$pField->getLabel().' (max)'] = $value;
                                } else {
                                        $output[$pField->getLabel()] = $value;
diff --git a/templates.dist/form/applicantform.php b/templates.dist/form/applicantform.php
index 5b6076d..a1fa5e0 100644
--- a/templates.dist/form/applicantform.php
+++ b/templates.dist/form/applicantform.php
@@ -58,6 +58,10 @@ foreach ( $pForm->getInputFields() as $input => $table ) {
                $searchcriteriaValues []= $line;
        }

+       if ($table === ''){
+               $addressValues []= $line;
+    }
+
 }

 if ($pForm->getFormStatus() !== \onOffice\WPlugin\FormPost::MESSAGE_SUCCESS) {
```

Calling `$pFieldsCollection->getFieldByKeyUnsafe()` is good enough for now, but could lead to strange issues later. It is better to know the module and call `getFieldByModuleAndName`, but the module of our message field is `''` and it'll make it harder later if the code has a bunch of `if ($field == 'message')` special cases.